fatiando/mesher/mesh.py

```diff
@@ -388,6 +388,117 @@ def copy(self):
         return cp.deepcopy(self)
 
 
+class PointMesh(object):
```

Comment:
Perhaps a better name would be `IrregularPointMesh` because we could also have a `PointMesh` that is like the `PrismMesh`.

Reply:
This class is actually a more generic version of `PointGrid` and could be used as its base class. The `PointGrid` would need to generate the x, y, z arrays and pass to this classes `__init__` method. This would remove all the duplicated code in `__getitem__` etc.

Reply:
I tried to create the class `IrregularPointMesh` as the base of `PointGrid`.